source/_posts/tips-for-beginners.md

```diff
@@ -0,0 +1,270 @@
+title: Tips for beginners
+date: 2015-06-10 03:05:00
+tags: tips
+---
+
+This blog posts comes with some useful tips and advices for beginners. 
+
+<!-- more -->
+
+## Data
+
+Initializing your data properties in Vue Model is safe and good for performance. Suppose you have following template:
+
+``` html
+<div id="demo">
+  <p v-class="green: validation.valid">{{message}}</p>
+  <input v-model="message">
+</div>
+```
+
+Recommended approach would be initializing data properties in you ViewModel, even if you need them to be empty/falsy:
+
+``` js
+new Vue({
+  el: '#demo',
+  
+  data: {
+    message: '',
+    validation: {
+      valid: false
+    }
+  }
+})
+```
+
+## Scope
+
+Every Vue.js component is a separate Vue child instance with it's own scope. It's important to understand how scopes work in Vue.js, especially when referring to ViewModel properties, methods from templates. Beginning from version 0.11, thumb of rule for scope in template is: 
+
+<p class="tip">if something appears in the parent template, it will be compiled in parent scope; if it appears in child template, it will be compiled in child scope.</p>
+
+Following snippets demonstrate *wrong* and *right* scope usage:
+
+``` js
+new Vue({
+  el: '#demo',
+
+  methods: {
+    parentMethod: function() {
+      return console.log('parent')
+    }
+  },
+
+  components: {
+    child: {
+      template: '<a v-on="click: parentMethod">click here</a>' // WRONG: you cannot access `parentMethod` from child instance (child component)
+      methods: {
+        childMethod: function() {
+          return console.log('child')
+        }
+      }
+    }
+  }
+})
+```
+
+``` html
+<div id="demo">
+  <!-- works, because this is parent template -->
+  <a v-on="click: parentMethod">click here</a>
+
+  <!-- does NOT work, because childMethod does not exist in this scope -->
+  <a v-on="click: childMethod"></a>
+</div>
+```
+
+## Communication between instances
```

Comment:
I'm thinking about whether we should list the use of events as best practice here. From what I've seen so far it is very easy to end up with complex state change flow when using events, because you have to simulate how events travel inside your component tree correctly to reason about what is happening.

Since I plan to release this post after 0.12, I think we should emphasis the usage of passing down methods as prop callbacks as the first recommended approach, with events filling in the gap of other scenarios.